pkg/util/winutil/service.go

```diff
@@ -0,0 +1,241 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+// +build windows
+
+package winutil
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"time"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	defaultServiceCommandTimeout = 10
+)
+
+var (
+	modadvapi32               = windows.NewLazyDLL("advapi32.dll")
+	procEnumDependentServices = modadvapi32.NewProc("EnumDependentServicesW")
+)
+
+type enumServiceState uint32
+
+func OpenSCManager(desiredAccess uint32) (*mgr.Mgr, error) {
+	h, err := windows.OpenSCManager(nil, nil, desiredAccess)
+	if h == 0 {
+		return nil, err
+	}
+	return &mgr.Mgr{Handle: h}, err
+}
+
+func OpenService(manager *mgr.Mgr, serviceName string, desiredAccess uint32) (*mgr.Service, error) {
+	h, err := windows.OpenService(manager.Handle, windows.StringToUTF16Ptr(serviceName), desiredAccess)
+	if err != nil {
+		return nil, err
+	}
+	return &mgr.Service{Name: serviceName, Handle: h}, err
+}
+
+// Start serviceName via SCM
+// Does not block until service is started
+// https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicea#remarks
+func StartService(serviceName string) error {
```

Comment:
I wonder if StartService cound be writtent in terms/logic of ControlService (with exception of Start and not control). But to have params timeout which by default could be 0.

Reply:
Could you elaborate on that? I think I see what you're saying, but not 100% clear.

Reply:
What I was saying that StartService can call ControlService and get benefit of timeout control. The only  enchancement would require is to have a bit to Start vs Control svc. Event that can be inferred from the req state.

Reply:
Ah okay, so there isn't a start control command, so I don't think this is an option.

Reply:
It is just a though and you do not to bother further or reply but imaginge if you want to add timeout to the StartService as well. You would endup implementing logic of what you have in ControlService minus a one line single API difference.  I was nitpicking of possibility of re-using ControlService-like functionality. Just to explain my thoughs above, May be it should be doControlService instead to keep external API correct. But again, it is just a thought.